src/golang/lib/engine/aq_engine.go

```diff
@@ -717,10 +724,65 @@ func (eng *aqEngine) cleanupWorkflow(ctx context.Context, workflowDag dag_utils.
 	}
 }
 
+func (eng *aqEngine) executeWithEngine(
+	ctx context.Context,
+	workflowDag dag_utils.WorkflowDag,
+	workflowName string,
+	engineConfig shared.EngineConfig,
+	storageConfig shared.StorageConfig,
+	workflowRunMetadata *WorkflowRunMetadata,
+	timeConfig *AqueductTimeConfig,
+	opExecMode operator.ExecutionMode,
+) error {
+	switch engineConfig.Type {
+	case shared.DatabricksEngineType:
+		jobConfig, err := operator.GenerateJobManagerConfig(
+			ctx,
+			engineConfig,
+			&storageConfig,
+			eng.AqPath,
+			eng.Vault,
+		)
+		if err != nil {
+			return errors.Wrap(err, "Unable to generate JobManagerConfig.")
+		}
+
+		jobManager, err := job.NewJobManager(jobConfig)
+		if err != nil {
+			return errors.Wrap(err, "Unable to create JobManager.")
+		}
+
+		databricksJobManager, ok := jobManager.(*job.DatabricksJobManager)
+		if !ok {
+			return errors.Wrap(err, "Unable to create DatabricksJobManager.")
+		}
+
+		return ExecuteDatabricks(
+			ctx,
+			workflowDag,
+			workflowName,
+			workflowRunMetadata,
+			timeConfig,
+			opExecMode,
+			databricksJobManager,
```

Comment:
As a follow up, why can't databrick be a separate engine implementing `execute()` with the same interface?